packages/dynamoose/test/types/Model.ts

```diff
@@ -110,3 +116,6 @@ UserTypedModel.update({"id": "foo"}, {
 UserTypedModel.update({"id": "foo"}, {
 	"$REMOVE":{"age":null}
 });
+
+// @ts-expect-error
+const shouldFailItemCustomMethodAccess = UserTypedModel.resetPassword();
```

Comment:
So are model methods outside the scope of this PR? This PR only deals with item methods it looks like?

Reply:
That's correct, I've decided to address them separately to keep scope focused. Happy to include model methods as well if that suits you?

Reply:
@tranhl Ideally I'd like to review both PRs before merging either of them. I don't want to be left in a state where one piece is done and people ask why the other piece wasn't implemented.

You can decide how to best structure it. I'd personally prefer 2 separate PRs so that it's easier to review and the context is easier to understand. Sadly I don't think GitHub has the concept of having one PR depend on another.